### PR TITLE
[New] Adding console.* polyfill

### PR DESCRIPTION
**Pull request by nicksorrentino**
While IE9 is still supported add a polyfill for console.* statement

@pnachum @ljharb

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,6 +10,9 @@ require('ima-babel6-polyfill');
 // document.contains polyfill
 require('./document-contains');
 
+// console.* polyfill for old browsers
```
Comment (lencioni) [inline index.js:13]: It would be nice if we could be more specific about which browsers.

**Review by lencioni — APPROVED** [review]

**Comment by nicksorrentino** [comment]
Okay updated the comment.

**Comment by lencioni** [comment]
Will you rebase this so it does not contain any merge commits?

**Comment by nicksorrentino** [comment]
okay rebased

**Review by ljharb — COMMENTED** [review]
I don't think we should do this.

The proper solution is to write tests for all our code, since console warnings fail tests - in other words, this only prevents breakage in code that's already broken (since code without tests is broken by default).

File: index.js [hunk 2]
```diff
@@ -10,6 +10,9 @@ require('ima-babel6-polyfill');
...
+require('console-polyfill')
```
Comment (ljharb) [inline index.js]: semicolon

**Review by ljharb — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (nicksorrentino) [reply index.js]: Whoops, when I rebased it reverted the fix I had locally.

**Comment by nicksorrentino** [comment]
Okay @ljharb based on the post mortem is seemed like a reasonable idea, it seemed like this was low effort and high-reward for the case when people eslint-disable.  Regardless if you prefer to keep the current behavior thats fine.

**Comment by ljharb** [comment]
@nicksorrentino hopefully nobody ever blindly does `eslint-disable` and only disables specific rules :-/

**Comment by nicksorrentino** [comment]
I mean more a specific rule so they can do a console.log or warn.  I think if we don't want this then we need to make console statements a error in eslint.  We do have some branches where they are only an eslint warn.

**Comment by pnachum** [comment]
I disagree with @ljharb on this one. 
1. It would be great if all of our code was tested, but it's not
2. For whatever reason, the warning from DLS did not break tests in monorail. 
3. If all else fails, as it did in my case (ie I saw no lint/test failures), it would be better to not break the page. 

Is there any downside to adding this?

**Comment by pnachum** [comment]
Also, doesn't `console.log` also break IE9, but not fail tests?

**Comment by ljharb** [comment]
Yes, those are good points that our tests don't fail on `console.log`, and that "if all else fails, it'd be better not to break the page".

**Review by ljharb — APPROVED** [review]
I still don't like needing to do this, but good points have been made.

**Comment by ljharb** [comment]
Released as v1.5.0

**Pull request merged**